src/onlyUpdateForProps.js

```diff
@@ -0,0 +1,8 @@
+import onlyUpdateForKeys from 'onlyUpdateForKeys';
+
+const onlyUpdateForProps = (BaseComponent) => {
+  const propKeys = Object.keys(BaseComponent.propTypes || {});
```

Comment:
Hmm, what should the behavior be if the component doesn't have any `propTypes`?

Perhaps we should warn? Otherwise the component will never update. I can't think of any situation where that would be intended behavior.